File: Lesson09_OOP/06_NotOO/CalculatorOO.java
/*
Copyright Ann Barcomb and Emily Marasco, 2022
Licensed under GPL v3
See LICENSE.txt for more information.
*/

public class CalculatorOO {

    private double one;
    private double two;

    public double add() { return one + two; }
    public double subtract() { return one - two; }
    public double multiply() { return one * two; }
    public double divide() { 
        if (two == 0) { return 0; }
        return one/two;
    }

    // Constructor
    public CalculatorOO(double one, double two) {
        this.one = one;
        this.two = two;
    }


    public static void main(String[] args) {
        double one = 9.0;
        double two = 0.1;
        var calculator = new CalculatorOO(one, two);
        System.out.println("Add: " + calculator.add());
        System.out.println("Subtract: " + calculator.subtract());
        System.out.println("Multiply: " + calculator.multiply());
        System.out.println("Divide: " + calculator.divide());
    }
}

File: Lesson09_OOP/06_NotOO/CalculatorOO.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""
class CalculatorOO:
    def __init__(self, one, two):
        self.one = one
        self.two = two

    def add(self):
        return self.one + self.two

    def subtract(self):
        return self.one - self.two

    def multiply(self):
        return self.one * self.two

    def divide(self):
        if self.two == 0:
            return 0
        return self.one / self.two

def main():
    one = 9.0
    two = 0.1
    calculator = CalculatorOO(one, two)
    print("Add:", calculator.add())
    print("Subtract:", calculator.subtract())
    print("Multiply:", calculator.multiply())
    print("Divide:", calculator.divide())

if __name__ == "__main__":
    main()
